### PR TITLE
Update frontend docs

### PR DESCRIPTION
**Pull request by FinalAngel**
- add naming conventions
- fix link to aldryn boilerplate bootstrap3
- add sekizai conventions

### CODE REVIEW

File: docs/reference/addons/template-structure.rst [hunk 1]
```diff
@@ -70,3 +70,48 @@ In this case the gallery provides two kind of templates, a standard version and
...
+-------
+
+It is recommended to use single-line sekizai declarations in order to merge
+duplicated entries for file injections::
+
+    {% load sekizai_tags %}
+    {% addtoblock "css" %}<link href="{% static 'css/theme.css' %}" rel="stylesheet">{% endaddtoblock %}
+    {% addtoblock "js" %}<script src="{% static 'libs/jquery.min.js' %}"></script>{% endaddtoblock %}
+
+Inline JavaScript should be avoided at all.
```
Comment (vxsx) [inline docs/reference/addons/template-structure.rst]: this line doesn't really provide the solution to the problem that was solved by using inline js, mb it makes sense to put some info on how to pass data from backend to js as well?

Comment (vxsx) [inline docs/reference/addons/template-structure.rst]: link to this mb, idk http://aldryn-boilerplate-bootstrap3.readthedocs.org/en/latest/codestyle/javascript.html#passing-data-to-components

**Comment by FinalAngel** [comment]
@vxsx added the link

File: docs/reference/addons/template-structure.rst [hunk 2]
```diff
@@ -70,3 +70,50 @@ In this case the gallery provides two kind of templates, a standard version and
...
+It is recommended to use single-line sekizai declarations in order to merge
+duplicated entries for file injections::
+
+    {% load sekizai_tags %}
+    {% addtoblock "css" %}<link href="{% static 'css/theme.css' %}" rel="stylesheet">{% endaddtoblock %}
+    {% addtoblock "js" %}<script src="{% static 'libs/jquery.min.js' %}"></script>{% endaddtoblock %}
+
+Inline JavaScript should be avoided at all. There is
+`additional reading material <http://aldryn-boilerplate-bootstrap3.readthedocs.org/en/latest/codestyle/javascript.html#passing-data-to-components>`_
+available if you want to get more examples of hwo to achieve this.
```
Comment (vxsx) [inline docs/reference/addons/template-structure.rst]: typo `hwo` -> `how`
also "achieve this" is not very descriptive, maybe change it to "on how to pass data from backend to the plugin" or something similar

File: docs/reference/addons/template-structure.rst [hunk 3]
```diff
@@ -26,7 +26,7 @@ look something like:
 
     {% extends "base.html" %}
     {% block content %}
-        <div class="addon-blog">
+        <div class="aldryn aldryn-blog">
```
Comment (mkoistinen) [inline docs/reference/addons/template-structure.rst]: It isn't clear what purpose, if any, having "aldryn" as a class everywhere.

Comment (FinalAngel) [inline docs/reference/addons/template-structure.rst]: thats the case for aldryn-newsblog already. The use is to have a generic name if styling needs to be done accross all addons. This is helpful for example to standardise font-size or overwrite default bootstrap components within the addons themselves.

**Comment by jmelett** [comment]
:+1:

File: docs/reference/addons/template-structure.rst (hunk 3 above)
Comment (mkoistinen) [inline docs/reference/addons/template-structure.rst]: Thanks @FinalAngel that's good to know, but I was expecting to see something to this affect in the documentation. This is indeed the part where you talk about standards for class names, but we're not addressing in the docs this very point about the `aldryn` class name and why it is there.

Comment (FinalAngel) [inline docs/reference/addons/template-structure.rst]: This is why I refer to the Aldryn Boilerplate Bootstrap3 documentation the entire philosophy is described. But if wanted, I can integrate some minor explanations here?

Comment (FinalAngel) [inline docs/reference/addons/template-structure.rst]: @mkoistinen see https://github.com/aldryn/aldryn-docs/commit/b4d22c2e58660ddcd0ff5964c95d06dde3bc05ec

**Comment by mkoistinen** [comment]
:+1:

File: docs/reference/addons/code-conventions.rst [hunk 4]
```diff
@@ -60,4 +60,13 @@ For full-page caching.
...
 .. TODO:: figure out some good rules for avoiding anonymous sessions.
           Should Apps be "certified" that they don't require anonymous sessions?
-          Configurable with setting? csrf_cookie? current language cookie?
\ No newline at end of file
+          Configurable with setting? csrf_cookie? current language cookie?
+
+Frontend
+--------
+
+There are many Boilerplates and equally as many code conventions and
```
Comment (evildmp) [inline docs/reference/addons/code-conventions.rst]: Actually I would strongly encourage more conformity than this, at least with accepted good practices, and to note that the more an application's code and documentation are in accord with the conventions adopted for official Aldryn packages, the better it is for the ecosystem and the more likely we are to endorse or promote it.

File: docs/reference/addons/template-structure.rst [hunk 5]
```diff
@@ -70,3 +70,50 @@ In this case the gallery provides two kind of templates, a standard version and
...
+In addition you can nest further if you are rendering a plugin or a list view
+etc.::
+
+    <div class="aldryn aldryn-events aldryn-events-latest">
+        <ul>
+            <li>...</li>
+        </ul>
+    </div>
+
+This allows frontend developers to customise your templates using css alone
```
Comment (evildmp) [inline docs/reference/addons/template-structure.rst]: css > CSS

File: docs/reference/boilerplates/code-conventions.rst [hunk 6]
```diff
@@ -1,9 +1,11 @@
 Code Conventions
 ================
 
-There are many Boilerplates and equally as many code conventions and styleguides. We will not go as far as to
-command you to use tabs over spaces, camelCase over under_score, but we might want you to redirect to our
-`aldryn-boilerplate <https://github.com/divio/divio-boilerplate>`_'s README.rst which offers some advice.
+There are many Boilerplates and equally as many code conventions and
```
Comment (evildmp) [inline docs/reference/boilerplates/code-conventions.rst]: Here too I would encourage conformity more strongly.

File: docs/reference/addons/code-conventions.rst (hunk 4 above)
Comment (FinalAngel) [inline docs/reference/addons/code-conventions.rst]: @evildmp yeah the guidelines in the frontend community vary largely depending on the libraries or frameworks you are using, we refer to our guidelines. That sentence was actually already there, I just copied it and reformatted to fit 80 lines.

File: docs/reference/boilerplates/code-conventions.rst (hunk 6 above)
Comment (FinalAngel) [inline docs/reference/boilerplates/code-conventions.rst]: here again, just reformatting

**Pull request merged**